quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Start.java

```diff
@@ -56,6 +58,10 @@ protected void doBeforeRun() {
         if (Environment.isDevProfile()) {
             throw new PropertyException(Messages.devProfileNotAllowedError(NAME));
         }
+        if (Configuration.getConfigValue(DB).getConfigSourceOrdinal() == 0) {
+          picocli.warn("Usage of the default value for the db option in the production profile is deprecated. Please explicitly set the db instead.");
+          throw new PropertyException(Messages.invalidOrMissingPropertyDB());
```

Comment:
This must be removed. Once we no longer wish to support h2 as the default production db, then we need to instead remove the default from the property mapper. 

That is scheduled for 27.0 I believe.